Fix testcase stub of `createModelDataFor`

machty opened the pull request:
`createModelDataFor` was the old name before it was changed
to `createRecordDataFor`

igorT commented:
@machty can you please add an assert in the code we are stubbing to ensure this time it is getting called.

machty commented:
@igorT while addressing your feedback I noticed these tests had an identical setup and what appeared to be a lot of leftover cruft from the original RFC implementation. It's much smaller now, performs the same tests, and the stub assertion you asked for is accomplished just by testing that that the constructed RecordData for a model has the expected type (which is what one of the tests was doing before).

igorT commented on packages/-ember-data/tests/integration/record-data/store-compatibility-test.js:
```diff
@@ -0,0 +1,164 @@
+  });
+});
+
+module('RecordData - Instantiation', function (hooks) {
```
Can we not split the two modules into two files please?

igorT commented:
@machty great work!

runspired commented:
@machty if you hapeen to have time to rebase this I think the tests will pass and I'd be happy to merge

runspired commented:
something is odd here, the more recent commits clearly show test refactoring that the github diff cannot seem to see :/
